barretenberg/cpp/src/barretenberg/api/api_ultra_honk.hpp

```diff
@@ -0,0 +1,311 @@
+#pragma once
+
+#include "barretenberg/api/acir_format_getters.hpp"
+#include "barretenberg/api/api.hpp"
+#include "barretenberg/api/get_bn254_crs.hpp"
+#include "barretenberg/api/init_srs.hpp"
+#include "barretenberg/api/write_prover_output.hpp"
+#include "barretenberg/common/map.hpp"
+#include "barretenberg/common/throw_or_abort.hpp"
+#include "barretenberg/dsl/acir_format/proof_surgeon.hpp"
+#include "barretenberg/dsl/acir_proofs/honk_contract.hpp"
+#include "barretenberg/dsl/acir_proofs/honk_zk_contract.hpp"
+#include "barretenberg/srs/global_crs.hpp"
+#include "barretenberg/ultra_vanilla_client_ivc/ultra_vanilla_client_ivc.hpp"
+
+namespace bb {
+
+extern std::string CRS_PATH;
+
+class UltraHonkAPI : public API {
+
+    template <typename Flavor, typename Circuit = Flavor::CircuitBuilder>
+    static Circuit _compute_circuit(const std::string& bytecode_path,
+                                    const std::string& witness_path,
+                                    const bool init_kzg_accumulator)
+    {
+        uint32_t honk_recursion = 0;
+        if constexpr (IsAnyOf<Flavor, UltraFlavor, UltraKeccakFlavor, UltraKeccakZKFlavor>) {
+            honk_recursion = 1;
+        } else if constexpr (IsAnyOf<Flavor, UltraRollupFlavor>) {
+            honk_recursion = 2;
+        }
+
+        // TODO(https://github.com/AztecProtocol/barretenberg/issues/1180): Don't init grumpkin crs when unnecessary.
+        init_grumpkin_crs(1 << CONST_ECCVM_LOG_N);
+
+        const acir_format::ProgramMetadata metadata{ .recursive = init_kzg_accumulator,
+                                                     .honk_recursion = honk_recursion };
+        acir_format::AcirProgram program{ get_constraint_system(bytecode_path, metadata.honk_recursion) };
+
+        if (!witness_path.empty()) {
+            program.witness = get_witness(witness_path);
+        }
+        auto circuit = acir_format::create_circuit<Circuit>(program, metadata);
+
+        return circuit;
+    }
+
+    template <typename Flavor>
+    static UltraProver_<Flavor> _compute_prover(const std::string& bytecode_path,
+                                                const std::string& witness_path,
+                                                const bool init_kzg_accumulator)
+    {
+        // Lambda function to ensure the builder gets freed before generating the vk. Vk generation requires initialing
```

Comment:
blocking request: 
confusing to read, can just be
```
// NOTE: During construction of UltraProver generate the VK (for debugging purposes). Peak memory is hit here due to pippenger runtime state construction.
// We therefore don't want to assign the circuit to a variable as the builder would be kept alive (increasing peak memory).
auto prover = UltraProver_<Flavor>{ _compute_circuit<Flavor>(bytecode_path, witness_path, init_kzg_accumulator)};
```

Reply:
Fixed, but to be clear this was not me